### PR TITLE
Refactoring Action for use with AKS Atlanta

### PR DESCRIPTION
**Pull request by tbarnes94**
* Did some reorganizing of code in run.ts, moved run.test.ts into /src, and put some helpers into helpers.ts in /src.

* Did some reorganizing of code in run.ts, moved run.test.ts into /src, and put some helpers into helpers.ts in /src.

* Grabbed the upstream integration tests and brought them here.  Removed bash script. Added validateKubectl.py to /test folder for integration tests.

* Ran npm run build

* Ran npm run build

* Updated on section for integration-tests.yml

* Removing ruby commands from integration tests yaml.

* Fixing discrepancies in integration test yaml.

* Fixing discrepancies in integration test yaml.

* Default to ubuntu-latest

* renamed python script according to workflow.

* renamed python script according to workflow.

* Fixing path parameters.

* Updated tsconfig.json

* Testing for int test failure.

* Validated that int tests work.

* Added new workflows.

Co-authored-by: Tommy Barnes <thbarnes@microsoft.com>

### CODE REVIEW

File: .github/workflows/release-pr.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      # these two only need to occur if the branch exists
+      - name: Checkout proper branch
+        if: ${{ steps.extract-branch-status.outputs.exists == 'true' }}
+        env:
+          BRANCH: releases/${{ github.event.inputs.release }}
+        run: git checkout ${BRANCH}
+      - name: Reset promotion branch
+        if: ${{ steps.extract-branch-status.outputs.exists == 'true' }}
+        run: |
+          git fetch origin master:master
```
Comment (OliverMKing) [inline .github/workflows/release-pr.yml:32]: these two lines need to change from master to main for this repo

**Review by OliverMKing — CHANGES_REQUESTED** [review]
Really good stuff :)! Made a few suggestions

**Review by davidgamero — APPROVED** [review]

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -1,19 +1,49 @@
-name: "Trigger Integration tests"
-on:
+name: Integration test for setup-kubectl-v1
```
[the remark this answers is not in the document]
Comment (tbarnes94) [reply .github/workflows/integration-tests.yml:1]: changed

File: .github/workflows/release-pr.yml (hunk 1 above)
Comment (tbarnes94) [reply .github/workflows/release-pr.yml:32]: changed

File: src/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        case 'Linux':
+            return util.format('https://storage.googleapis.com/kubernetes-release/release/%s/bin/linux/%s/kubectl', version, arch);
+
+        case 'Darwin':
+            return util.format('https://storage.googleapis.com/kubernetes-release/release/%s/bin/darwin/%s/kubectl', version, arch);
+
+        case 'Windows_NT':
+        default:
+            return util.format('https://storage.googleapis.com/kubernetes-release/release/%s/bin/windows/%s/kubectl.exe', version, arch);
+
```
[the remark this answers is not in the document]
Comment (tbarnes94) [reply src/helpers.ts:23]: changed

**Review by OliverMKing — APPROVED** [review]

**Pull request merged**